api/managementpb/dbaas/psmdb_cluster.proto

```diff
@@ -0,0 +1,211 @@
+  PSMDB_CLUSTER_STATE_CHANGING = 1;
+  // PSMDB_CLUSTER_STATE_READY represents a cluster without pending changes (ready).
+  PSMDB_CLUSTER_STATE_READY = 2;
+  // PSMDB_CLUSTER_STATE_FAILED represents a failed cluster (error).
+  PSMDB_CLUSTER_STATE_FAILED = 3;
+  // PSMDB_CLUSTER_STATE_DELETING represents a cluster being deleting.
+  PSMDB_CLUSTER_STATE_DELETING = 4;
+}
+
+// PSMDBClusterParams represents PSMDB cluster parameters that can be updated.
+message PSMDBClusterParams {
+  // ReplicaSet container parameters.
+  message ReplicaSet {
+    // Requested compute resources.
+    ComputeResources compute_resources = 1;
+  }
+  // Cluster size.
+  int32 cluster_size = 1 [
+    (validator.field) = {
+      int_gt: 0
+    }
+  ];
+  // ReplicaSet container parameters.
+  ReplicaSet replicaset = 2 [
+    (validator.field) = {
+      msg_exists: true
+    }
+  ];
+}
+
+message ListPSMDBClustersRequest {
+  // Kubernetes cluster name.
+  string kubernetes_cluster_name = 1 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+}
+
+message ListPSMDBClustersResponse {
+  // Cluster represents PSMDB cluster information.
+  message Cluster {
+    // Cluster name.
+    string name = 1;
+    // Cluster state.
+    PSMDBClusterState state = 2;
+    // Currently running operation, if any.
+    RunningOperation operation = 3;
+    // Cluster parameters.
+    PSMDBClusterParams params = 4;
+  }
+  // PSMDB clusters information.
+  repeated Cluster clusters = 1;
+}
+
+message GetPSMDBClusterRequest {
+  // Kubernetes cluster name.
+  string kubernetes_cluster_name = 1 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+  // PSMDB cluster name.
+  string name = 2 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+}
+
+message GetPSMDBClusterResponse {
+  // PSMDBCredentials is a credentials to connect to PSMDB.
+  message PSMDBCredentials {
+    // MongoDB username.
+    string username = 5;
+    // MongoDB password.
+    string password = 6;
+    // MongoDB host.
+    string host = 7;
+    // MongoDB port.
+    int32 port = 8;
+  }
+  // Cluster state.
+  PSMDBClusterState state = 1;
+  // Currently running operation, if any.
+  RunningOperation operation = 2;
+  // Cluster parameters.
+  PSMDBClusterParams params = 3;
+  // Connection credentials.
+  PSMDBCredentials connection_credentials = 4;
+}
+
+message CreatePSMDBClusterRequest {
+  // Kubernetes cluster name.
+  // a DNS-1035 label must consist of lower case alphanumeric characters or '-',
+  // start with an alphabetic character, and end with an alphanumeric character
+  // (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')
+  string kubernetes_cluster_name = 1 [
+    (validator.field) = {
+      regex: "^[a-z]([-a-z0-9]*[a-z0-9])?$"
+    }
+  ];
+  // PSMDB cluster name.
+  string name = 2 [
+    (validator.field) = {
+      string_not_empty: true
+    }
```

Comment:
Please swap validation of name (should be DNS-1035) and kubernetes_cluster_name (should be not empty).

Reply:
Marking as resolved